pages/visualizations/overview/commits_over_time.py

```diff
@@ -118,39 +119,72 @@ def toggle_popover_2(n, is_open):
 def create_commits_over_time_graph(repolist, timer_pings, interval):
     logging.debug("COT - PONG")
 
+    # start timer
+    start = time.perf_counter()
+
+    # get the results from the job queue if they're available
     ready, results, graph_update, interval_update = handle_job_state(jm, cmq, repolist)
     if not ready:
         # set n_intervals to 0 so it'll fire again.
         return graph_update, interval_update
 
-    logging.debug("COMMITS_OVER_TIME_VIZ - START")
-    start = time.perf_counter()
+    # default time values for figure
+    x_r, x_name, hover, period = get_graph_time_values(interval)
 
-    df_commits = pd.DataFrame(results)
+    # process raw data into plotable format
+    df = process_data(results, period)
 
-    # reset index to be ready for plotly
-    df_commits = df_commits.reset_index()
+    # create figure from processed data
+    fig = create_figure(df, x_r, x_name, hover, interval)
 
-    # time values for graph
-    x_r, x_name, hover, period = get_graph_time_values(interval)
+    logging.debug(f"COMMITS_OVER_TIME_VIZ - END - {time.perf_counter() - start}")
+    return fig, dash.no_update
+
+
+def create_figure(df: pd.DataFrame, x_r, x_name, hover, interval):
+
+    # create figure
+    fig = px.bar(data_frame=df, x="period", y="counts", range_x=x_r, labels={"x": x_name, "y": "Commits"})
+
+    # set figure styling
+    fig.update_layout(
+        xaxis_title=x_name,
+        yaxis_title="Number of Commits",
+        margin_b=40,
+        margin_r=20,
+        # fill in the height of the graph, max + 10%
+        yaxis_range=[0, (df["counts"].max() * 1.10)],
+    )
+
+    # set the axes to include the rangeslider
+    fig.update_xaxes(
+        showgrid=True,
+        ticklabelmode="instant",
+        dtick=interval,
+        rangeslider_yaxis_rangemode="match",
+    )
+
+    # set the hover template
+    fig.update_traces(hovertemplate=hover + "<br>Commits: %{y}<br>")
+
+    return fig
+
+
+def process_data(results, period):
```

Comment:
Whats the reasoning for breaking down into another function?

Reply:
/should this be in graph utils if its planned to be generic to be used in multiple graphs?